### PR TITLE
fix(socket sink): Allow socket sink to accept metrics.

### PR DESCRIPTION
**Pull request by nichtverstehen**
Closes https://github.com/vectordotdev/vector/issues/20756

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
Great, thank you @nichtverstehen !

One request, could we add a changelog entry describing the change for Vector users? See https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md for details.

**Comment by nichtverstehen** [comment]
@jszwedko Thanks for the quick review. Added a changelog entry.

File: changelog.d/20930_socket_sink_accepts_metrics.fix.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Socket sink can now accept metric events (when using `native_json` codec).
```
Comment (jszwedko) [inline changelog.d/20930_socket_sink_accepts_metrics.fix.md:1]:
```suggestion
Socket sink can now accept metric events when using codecs that support decoding metrics.
```

I think the full list is actually `native_json`, `native`, and, recently added, `influxdb`. I'd just leave that off though.

**Review by jszwedko — COMMENTED** [review]
One note about the changelog, otherwise 👍

File: changelog.d/20930_socket_sink_accepts_metrics.fix.md (hunk 1 above)
Comment (jorgehermo9) [reply changelog.d/20930_socket_sink_accepts_metrics.fix.md:1]: wouldn't it be "when using codecs that supports *encoding* metrics" instead?

Comment (nichtverstehen) [reply changelog.d/20930_socket_sink_accepts_metrics.fix.md]: Thanks. Applied your suggestion @jorgehermo9

File: src/sinks/socket.rs [hunk 2]
```diff
@@ -276,6 +279,46 @@ mod test {
...
+        .expect("Running sink failed");
+
+        // Wait for output to connect
+        receiver.connected().await;
+
+        let output = receiver.await;
+        assert_eq!(expected.len(), output.len());
+        for (source, received) in expected.iter().zip(output) {
+            let json = serde_json::from_str::<Value>(&received).expect("Invalid JSON");
+            println!("VERSION {:?}", json);
```
Comment (jorgehermo9) [inline src/sinks/socket.rs:315]: forgot to delete this print?
```suggestion
```

Comment (nichtverstehen) [reply src/sinks/socket.rs]: yes 🤦‍♂️. removed.

File: changelog.d/20930_socket_sink_accepts_metrics.fix.md (hunk 1 above)
Comment (jszwedko) [reply changelog.d/20930_socket_sink_accepts_metrics.fix.md]: Doh, yes, _encoding_. Thanks for the catch @jorgehermo9

**Review by jszwedko — APPROVED** [review]

**Comment by nichtverstehen** [comment]
Made changes required by a check. @jszwedko could you please reapprove?

File: src/sinks/socket.rs [hunk 3]
```diff
@@ -276,6 +279,45 @@ mod test {
         }
     }
 
+    #[tokio::test]
+    async fn metrics_socket() {
```
Comment (jszwedko) [inline src/sinks/socket.rs:283]: This added test won't be able to run on Windows. Do you mind gating it to run only on *nix? `#[cfg(target_family = "unix")]`

Comment (nichtverstehen) [reply src/sinks/socket.rs:284]: Done. (Added `#cfg(unix)` which seem to be used elsewhere). Let's see if windows is happy.

**Comment by nichtverstehen** [comment]
Sigh. Made imports conditional.

**Comment by nichtverstehen** [comment]
Added a platform guard for the symbol that became unused after guarding another unused symbol.

Is there a way to run the windows linter ahead of time?

**Comment by jszwedko** [comment]
> Added a platform guard for the symbol that became unused after guarding another unused symbol.
> 
> Is there a way to run the windows linter ahead of time?

Thanks! It doesn't run on PRs until the merge queue because it is more expensive than the other checks and doesn't fail often in isolation (i.e. if it fails the Linux one, which does run on PRs, would fail too).

**Pull request merged**